docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-front-end.md

```diff
@@ -151,6 +155,201 @@ export default {
+:::
+
+To migrate to Strapi v4, depending on what your Strapi v3 plugin does, use the following table to find the appopriate Settings API method to use, and click on the method name to go to its dedicated documentation:
+
+| Action     | Method |
+|-----|----|
+| Create a new settings section<br/> and define new links to include in this section | [`createSettingsSection()`](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html#createsettingsection) |
+| Add link(s) to an existing settings section:<ul><li>a single link</li><li>multiple links</li></ul> | <br/><ul><li>[`addSettingsLink()`](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html#addsettingslink)</li><li>[`addSettingsLinks()`](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html#addsettingslinks)</li></ul> |
+
+::: details Example of creating a new settings section
+
+```js
+// path: ./src/plugins/my-plugin/admin/src/index.js
+
+import getTrad from './utils/getTrad';
+
+register(app) {
+  // Create the plugin's settings section
+  app.createSettingSection(
+    // created section
+    {
+      id: pluginId,
+      intlLabel: {
+        id: getTrad('Settings.section-label'),
+        defaultMessage: 'My plugin settings',
+      },
+    },
+    // links
+    [
+      {
+        intlLabel: {
+          id: 'settings.page',
+          defaultMessage: 'Setting page 1',
+        },
+        id: 'settings',
+        to: `/settings/my-plugin/`,
+        Component: async () => {
+          const component = await import(
+            /* webpackChunkName: "my-plugin-settings-page" */ './pages/Settings'
+          );
+
+          return component;
+        },
+        permissions: [],
+      },
+
+    ]
+  );
+
+  app.registerPlugin({
+    id: pluginId,
+    name,
+  });
+},
+```
+
+:::
+
+## Adding reducers
+
+::: strapi v3/v4 comparison
+
+A Strapi v3 plugin adds reducers by exporting a `reducers` property during the plugin registration.
+
+In Strapi v4, a plugin adds reducers programmatically using the [Reducers API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#reducers-api).
+:::
+
+To migrate to Strapi v4, make sure reducers are added programmatically with the `addReducers()` method.
+
+<!-- ? did the reducers format change in v4? since it's not documented in Strapi v3, I can't compare… 😅  ? -->
+
+::: details Example of adding reducers
+
+```js
+// path: ./src/plugins/my-plugin/admin/src/index.js
+
+import myReducer from './components/MyCompo/reducer';
+import myReducer1 from './components/MyCompo1/reducer';
+import pluginId from './pluginId';
+
+const reducers = {
+  [`${pluginId}_reducer`]: myReducer,
+  [`${pluginId}_reducer1`]: myReducer1,
+};
+
+export default {
+  register(app) {
+    app.addReducers(reducers);
+
+    app.registerPlugin({
+      id: pluginId,
+      initializer: Initializer,
```

Comment:
Here I would remove the initializer and the isReady keys are they are optional and create noises.
IMO we should only talk about it in the Initializer documentation.
WDYT?

Reply:
Excellent idea, thanks!